src/container.c

```diff
@@ -487,6 +536,11 @@ static int hyper_container_init(void *data)
 		goto fail;
 	}
 
+	if (container_setup_init_layer(container) < 0) {
+		fprintf(stderr, "container sets up init layer failed\n");
+		goto fail;
+	}
+
```

Comment:
it should be called before container_setup_mount() which uses the created mountpoints.